tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -0,0 +1,305 @@
+        Builder::default()
+    }
+}
+
+// This, like the MakeWriter block, needs to be a seperate impl block because we're
+// overriding the `E` type parameter with `E2`.
+impl<S, N, E, W> Builder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Sets a [FormatEvent<S, N>].
+    pub fn with_event_formatter<E2>(self, e: E2) -> Builder<S, N, E2, W>
+    where
+        E2: FormatEvent<S, N> + 'static,
+    {
+        Builder {
+            fmt_fields: self.fmt_fields,
+            fmt_event: e,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+// this needs to be a seperate impl block because we're re-assigning the the W2 (make_writer)
+// type paramater from the default.
+impl<S, N, E, W> Builder<S, N, E, W> {
+    /// Sets a [MakeWriter] for spans and events.
+    pub fn with_writer<W2>(self, make_writer: W2) -> Builder<S, N, E, W2>
+    where
+        W2: MakeWriter + 'static,
+    {
+        Builder {
+            fmt_fields: self.fmt_fields,
+            fmt_event: self.fmt_event,
+            make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S, N, E, W> Builder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Builds a [FmtLayer] infalliably.
+    pub fn build(self) -> FmtLayer<S, N, E, W> {
+        FmtLayer {
+            make_writer: self.make_writer,
+            fmt_fields: self.fmt_fields,
+            fmt_event: self.fmt_event,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S> Default for FmtLayer<S>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+{
+    fn default() -> Self {
+        Builder::default().build()
+    }
+}
+
+impl<S> Default for Builder<S> {
+    fn default() -> Self {
+        Self {
+            fmt_fields: format::DefaultFields::default(),
+            fmt_event: format::Format::default(),
+            make_writer: io::stdout,
+            _inner: PhantomData,
+        }
+    }
+}
+
+impl<S, N, E, W> FmtLayer<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    #[inline]
+    fn make_ctx<'a>(&'a self, ctx: Context<'a, S>) -> FmtContext<'a, S, N> {
+        FmtContext {
+            ctx,
+            fmt_fields: &self.fmt_fields,
+        }
+    }
+}
+
+// === impl Formatter ===
+
+/// A formatted representation of a span's fields stored in its extensions.
+///
+/// By storing [FormattedFields] instead of a [String] directly,
+/// [FmtLayer] is able to be more defensive about other layers
+/// accidentally a span's extensions.
```

Comment:
these links need to be fixed...also, i think i left a comment on this earlier about a possible rewording? maybe it was lost in the move